src/Microsoft.AspNet.Mvc.Core/Rendering/Html/DefaultHtmlGenerator.cs

```diff
@@ -990,6 +1085,37 @@ internal static object GetModelStateValue(ViewContext viewContext, string key, T
             return UnobtrusiveValidationAttributesGenerator.GetValidationAttributes(clientRules);
         }
 
+        private static Enum ConvertEnumFromInteger(object value, Type targetType)
+        {
+            try
+            {
+                return (Enum)Enum.ToObject(targetType, value);
+            }
+            catch (Exception exception)
```

Comment:
Seems like the right fix though? Why not use a feature that's specifically designed for this scenario?